### PR TITLE
User api support

### PR DESCRIPTION
**Pull request by brenttyler**
Hello, Uwe! I have a home-grown application that relies on User favorites. A year or two ago, TVDB changed their API under me and instead of rewriting the interface myself, I found your package. Sadly it was missing the implementation for this path of the API so I took the liberty to add it myself. It's been working just fine for me for a few months so I decided I should finally check it in and see if you'd like to add it to the main.

There is a small concern in the JUnit testing in that one of the tests (TheTvdbuserTest.test_user() ) asserts that the logged-in user matches TestData.USER_NAME based on the value in BaseTestCase.API_KEY. Right now that value is set to blank so it will fail. But if you change it to match your user name (which TVDB gives back based on the your API KEY) then it will work. I considered creating a dummy TVDB account that can be used for these JUnit testing with its own API key, but didn't want to do that myself in case it was against their TOS.

Anyway, feel free to check it out and let me know if you have any questions or concerns. Thanks for making this easy for me!

Cheers,
Brent

### CODE REVIEW

**Comment by UweTrottmann** [comment]
Thank you! Based on a first look is great and well tested. The only thing I would ask is that you change your code style settings to match the existing code and then reformat your contributed code.

I'll have a look about the test account. The current test API key is (I believe) associated with my personal account, so we'll see.

**Comment by brenttyler** [comment]
Sure thing! I thought I had followed your code style when making these changes, but I'll go back over them this weekend and make sure. Can you point me to something specific at which to look?

File: src/main/java/com/uwetrottmann/thetvdb/TheTvdb.java [hunk 1]
```diff
@@ -139,5 +141,12 @@ public TheTvdbSearch search() {
     public TheTvdbUpdated updated() {
         return getRetrofit().create(TheTvdbUpdated.class);
     }
+    
+    /**
+     * Routes for handling user data.
+     */
+    public TheTvdbUser user() {
+    	return getRetrofit().create(TheTvdbUser.class);
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/thetvdb/TheTvdb.java:149]: Some code style examples: indentation 2 vs 4 spaces.

File: src/main/java/com/uwetrottmann/thetvdb/entities/User.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.uwetrottmann.thetvdb.entities;
+
+public class User
+{
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/thetvdb/entities/User.java:4]: Parentheses on new line.

File: src/main/java/com/uwetrottmann/thetvdb/entities/UserFavorites.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.uwetrottmann.thetvdb.entities;
+
+import java.util.List;
+
+public class UserFavorites {
+	public List<String> favorites;
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/thetvdb/entities/UserFavorites.java:6]: Indentation using tabs vs. spaces.

**Comment by brenttyler** [comment]
Pushed a code formatting change. I think I got it all. Let me know!

**Comment by UweTrottmann** [comment]
Thanks again! I rebased and merged separately including some simplifications.

**Pull request closed without merging**